backend/src/main/java/com/banorte/account/transactions/service/TransactionsService.java

```diff
@@ -0,0 +1,43 @@
+package com.banorte.account.transactions.service;
+
+import com.banorte.account.transactions.repository.TransactionsRepository;
+import com.banorte.account.transactions.repository.entity.TransactionEntity;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Service
+public class TransactionsService {
+
+  private final TransactionsRepository transactionsRepository;
+
+  public TransactionsService(
+      TransactionsRepository transactionsRepository
+  ) {
+    this.transactionsRepository = transactionsRepository;
+  }
+
+  public Flux<TransactionEntity> getAll(TransactionEntity.TransactionType type) {
+    return switch (type) {
```

Comment:
return (type != null) ? this.transactionRepository.findByType(type) : this.transactionRepository.findAll(); o uso de Optionals

Reply:
queda mas enredado usando el ternary y el optional, justo por todas las validaciones que se tienen que hacer, me parecio mas adecuado y entendible el switch